exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/aggregate/HashAggTemplate.java

```diff
@@ -306,14 +685,29 @@ public AggOutcome doWork() {
 
           while (true) {
             // Cleanup the previous batch since we are done processing it.
+            long pre = allocator.getAllocatedMemory();
             for (VectorWrapper<?> v : incoming) {
               v.getValueVector().clear();
             }
+            long beforeAlloc = allocator.getAllocatedMemory();
+
```

Comment:
This function seems overly long: very hard to follow. Perhaps break out each nested loop into a function. Since the functions are called once per batch, the cost of a function call is lost in the noise of operating the iterator hierarchy. The gain, however, is lower mental cost to understand what's happening.